pkg/rke2/registries.go

```diff
@@ -45,85 +47,96 @@ func GenerateRegistries(rke2ConfigRegistry RegistryScope) (*Registry, []bootstra
 	}
 
 	for configName, regConfig := range rke2ConfigRegistry.Registry.Configs {
-		tlsSecret := corev1.Secret{}
-		authSecret := corev1.Secret{}
-
-		err := rke2ConfigRegistry.Client.Get(
-			rke2ConfigRegistry.Ctx,
-			types.NamespacedName{
-				Name:      regConfig.TLS.TLSConfigSecret.Name,
-				Namespace: regConfig.TLS.TLSConfigSecret.Namespace,
-			},
-			&tlsSecret,
-		)
-		if err != nil {
-			rke2ConfigRegistry.Logger.Error(err, "TLS Config Secret for the registry was not found!")
-
-			return &Registry{}, []bootstrapv1.File{}, err
+		registryConfig := RegistryConfig{}
+
+		if regConfig.TLS == (bootstrapv1.TLSConfig{}) && regConfig.TLS != (bootstrapv1.TLSConfig{}) {
+			return &Registry{}, []bootstrapv1.File{}, fmt.Errorf("need to specify either authSecret or tls.tlsConfig")
 		}
 
-		for _, secretEntry := range []string{"tls.crt", "tls.key", "ca.crt"} {
-			if tlsSecret.Data[secretEntry] == nil {
-				rke2ConfigRegistry.Logger.Error(err, "TLS Config Secret for the registry is missing entries!", "secret-missing-entry", secretEntry)
+		if regConfig.TLS != (bootstrapv1.TLSConfig{}) {
+			tlsSecret := corev1.Secret{}
+
+			err := rke2ConfigRegistry.Client.Get(
+				rke2ConfigRegistry.Ctx,
+				types.NamespacedName{
+					Name:      regConfig.TLS.TLSConfigSecret.Name,
+					Namespace: regConfig.TLS.TLSConfigSecret.Namespace,
+				},
+				&tlsSecret,
+			)
+			if err != nil {
+				rke2ConfigRegistry.Logger.Error(err, "TLS Config Secret for the registry was not found!")
```

Comment:
Perhaps we can check for a specific `NotFound` error here, so that we have clearer logging based on the type of error:
```
if apierrors.IsNotFound(err) {
    rke2ConfigRegistry.Logger.Error(err, "TLS Config Secret for the registry was not found!")
}
```
And a different log message for a failed `Get` request. What do you think?

Reply:
I did this instead 
```
			if err != nil {
				if apierrors.IsNotFound(err) {
					rke2ConfigRegistry.Logger.Error(err, "TLS Config Secret for the registry was not found!")
				} else {
					rke2ConfigRegistry.Logger.Error(err, "Error fetching TLS Config Secret")
				}

				return &Registry{}, []bootstrapv1.File{}, err
			}```